samples/CMakeLists.txt

```diff
@@ -95,83 +73,70 @@ if (CONFIG_SECURE_BOOT)
-      message("Adding 'hci_rpmsg' firmware as child image since "
-      "CONFIG_BT_RPMSG_NRF53 is set to 'y'")
-    endif()
-
-    if (CONFIG_BT_LL_SOFTDEVICE_DEFAULT)
-      add_overlay_config(
-        ${NETCORE_IMAGE}
-        ${NRF_DIR}/subsys/bluetooth/controller/bt_ll_softdevice_hci_rpmsg.conf
-        )
-    endif()
+if (CONFIG_NCS_AUTO_INCLUDE_NET_CORE_CHILD_IMAGE
+    OR CONFIG_NCS_CHILD_IMAGE_SAMPLE_EMPTY_APP_CORE)
 
-    network_core_firmware_update_enable(${NETCORE_IMAGE})
+  if(CONFIG_NCS_CHILD_IMAGE_SAMPLE_EMPTY_APP_CORE)
+    set(CHILD_IMAGE_NAME "empty_app_core")
+    set(CHILD_IMAGE_PATH "${NRF_DIR}/samples/nrf5340/empty_app_core")
+
+  elseif(CONFIG_NCS_CHILD_IMAGE_SAMPLE_MULTIPROTOCOL_RPMSG)
+    set(CHILD_IMAGE_NAME "multiprotocol_rpmsg")
+    set(CHILD_IMAGE_PATH "${NRF_DIR}/samples/nrf5340/multiprotocol_rpmsg")
+
+  elseif(CONFIG_NCS_CHILD_IMAGE_SAMPLE_HCI_RPMSG)
+    set(CHILD_IMAGE_NAME "hci_rpmsg")
+    set(CHILD_IMAGE_PATH "${ZEPHYR_BASE}/samples/bluetooth/hci_rpmsg")
+
+  elseif(CONFIG_NCS_CHILD_IMAGE_SAMPLE_802154_RPMSG)
+    set(CHILD_IMAGE_NAME "802154_rpmsg")
+    set(CHILD_IMAGE_PATH "${ZEPHYR_BASE}/samples/boards/nrf/ieee802154/802154_rpmsg")
+
+  elseif(CONFIG_NCS_CHILD_IMAGE_SAMPLE_RPC_HOST)
+    set(CHILD_IMAGE_NAME "rpc_host")
+    set(CHILD_IMAGE_PATH "${NRF_DIR}/samples/bluetooth/rpc_host")
 
-    add_child_image(
-      NAME ${NETCORE_IMAGE}
-      SOURCE_DIR ${NETCORE_IMAGE_PATH}
-      DOMAIN CPUNET
-      BOARD ${CONFIG_DOMAIN_CPUNET_BOARD})
   else()
-    message(FATAL_ERROR
-      "SoC not supported for including hci_rpmsg as child image")
+    set(CHILD_IMAGE_NAME "hello_world")
+    set(CHILD_IMAGE_PATH "${ZEPHYR_BASE}/samples/hello_world")
   endif()
-elseif (CONFIG_NRF_802154_SER_HOST)
-  if (CONFIG_SOC_NRF5340_CPUAPP)
-    message("Adding '802154_rpmsg' firmware as child image since "
-    "CONFIG_NRF_802154_SER_HOST is set to y")
 
-    add_child_image(
-      NAME 802154_rpmsg
-      SOURCE_DIR ${ZEPHYR_BASE}/samples/boards/nrf/ieee802154/802154_rpmsg
-      DOMAIN CPUNET
-      BOARD ${CONFIG_DOMAIN_CPUNET_BOARD}
-    )
+  if(CONFIG_SOC_NRF5340_CPUAPP)
+    set(CHILD_IMAGE_DOMAIN "CPUNET")
   else()
-    message(FATAL_ERROR
-      "Board ${BOARD} not supported for including '802154_rpmsg' as child image")
+    set(CHILD_IMAGE_DOMAIN "CPUAPP")
   endif()
-endif()
-
-if (CONFIG_NCS_SAMPLE_EMPTY_APP_CORE_CHILD_IMAGE)
-  message("Adding 'empty_app_core' firmware as child image since "
-    "CONFIG_NCS_SAMPLE_EMPTY_APP_CORE_CHILD_IMAGE is set to 'y'")
-
-  add_child_image(
-    NAME empty_app_core
-    SOURCE_DIR ${NRF_DIR}/samples/nrf5340/empty_app_core
-    DOMAIN CPUAPP
-    BOARD ${CONFIG_DOMAIN_CPUAPP_BOARD})
-endif()
 
-if (CONFIG_BT_RPC)
-  if (CONFIG_SOC_NRF5340_CPUAPP)
-    message("Adding 'rpc_host' firmware as child image since "
-      "CONFIG_BT_RPC is set to 'y'")
+  if (CONFIG_NRF53_UPGRADE_NETWORK_CORE)
```

Comment:
What happened to the mcuboot dependency ?
https://github.com/nrfconnect/sdk-nrf/blob/27502d7684c8a6a1a9d72491a9cb0979cc9fbd3b/samples/CMakeLists.txt#L10